src/app/api/workflow-states/workflowStates.service.ts

```diff
@@ -11,14 +11,33 @@ class WorkflowStatesService extends BaseService {
     const policyGate = new PoliciesService(this.user)
     policyGate.authorize(UserAction.Read, Resource.Tasks)
 
-    const filters = { where: { workspaceId: this.user.workspaceId } }
-    let workflowStates = await this.db.workflowState.findMany(filters)
+    let workflowStates: WorkflowStateResponse[] = await this.db.$queryRaw`
+    SELECT * FROM "WorkflowStates"
+    WHERE "workspaceId" = ${this.user.workspaceId}
+    ORDER BY 
+      CASE 
+        WHEN "key" = 'todo' THEN 1
+        WHEN "key" = 'inProgress' THEN 2
+        WHEN "key" = 'completed' THEN 3
+        ELSE 4
+      END;
```

Comment:
Since we are using a raw query and have soft delete implemented prisma extensions will not work. You also have to add a case `WHERE "deletedAt" IS NULL`

Reply:
Can you please look into if Prisma implements case when natively? This way we can prevent writing a raw query.